docs/source/user_guide/getting_started.rst

```diff
@@ -576,10 +575,77 @@ within containers. The disadvantage is that you would need to edit the code to
 change the default values and hence site or project specific defaults are not
 suited for this purpose. The second way of specifying default are via ``layers``.
 
+Two issues arise from the behaviour of ``ConfigSuite``; what if a valid value is
+``None``? And how can default values be configured to something else then
+``None``? We try to facilitate both behaviours and address some solutions in
+the next sections.
+
 Note that no element should be both required and have a given ``Default`` value.
 
+Allow None
+~~~~~~~~~~
+
+For certain configurations it may be reasonable for the user to provide
+``None`` as the value. The ``None`` is a specific value in the eyes of
+``ConfigSuite`` as it is also the value when a value hasn't been provided.
+As such, the schema must be specifically declared to allow the user to set
+``None`` in order for a configuration to be valid if it contains ``None``.
+
+The ``AllowNone`` type is only valid for ``BasicType``'s, not for containers.
+Setting ``AllowNone`` to ``True`` for anything but ``BasicType`` will result
+in an invalid schema.
+
+Let us see how the owner section of the ``cars`` schema could be configured
+in order for a configuration with ``None`` to pass.
+
+.. testcode:: [allownone]
+
+    import configsuite
+    from configsuite import types
+    from configsuite import MetaKeys as MK
+
+    schema = {
+        MK.Type: types.NamedDict,
+        MK.Content: {
+            "owner": {
+                MK.Type: types.NamedDict,
+                MK.Content: {
+                    "name": {MK.Type: types.String},
+                    "credit": {
+                        MK.Type: types.Number,
+                        MK.AllowNone: True,
+                    },
+                    "insured": {MK.Type: types.Bool},
+                },
+            },
+        },
+    }
+
+    config = {
+        "owner": {
+                "name": "Scrooge",
+                "credit": None,
+                "insured": False,
+        },
+    }
+
+    suite = configsuite.ConfigSuite(config, schema)
+    assert suite.valid
+
+    owner = suite.snapshot.owner
+    print("{} has a credit of {}".format(owner.name, owner.credit))
+
+.. testoutput:: [allownone]
+
+    Scrooge has a credit of None
+
+An important implication of allowing ``None`` is that it is impossible to know
```

Comment:
You will never be able to separate between this anymore. If you allow `None` there is no way to tell whether it comes from the user or as a "default" default value. Hence, this section but the last line should be removed...

Reply:
>You will never be able to separate between..

I don't understand. You weren't able to separate between these from before either. What do you mean by `anymore`? After the merging of https://github.com/equinor/configsuite/pull/70?

If any I would add the following after the default has been merged. 

>As such ``ConfigSuite`` requires that any entity that is not required must also allow ``None``, unless the default value has been explicitly set to something else than ``None``.

(although I do think this setup is starting to get pretty complex to understand.. :) )

Reply:
> An important implication of allowing ``None`` is that it is impossible to know whether the value in the configuration was set to ``None`` explicitly, or if it is just a result of a missing configuration item.

I just think that this indicates that it is different if you do not allow `None`, which it should not be. Hence, I would prefer if we just remove the sentence altogether.

Reply:
Alright, so we agree that the statement is correct, it's more that how it's formulated could be interpreted as the "default" behaviour would differ depending on if `None` is allowed.

I just wanted to add a reasoning as to why ConfigSuite were strict about this, I don't think it's obvious (at least unless you specifically think about it..).

As long as it can be ambiguous I'll remove it.